managed-ledger/src/main/java/org/apache/bookkeeper/mledger/LedgerOffloader.java

```diff
@@ -85,6 +181,34 @@
                                     UUID uid,
                                     Map<String, String> extraMetadata);
 
+    /**
+     * Begin offload the passed in ledgers to longterm storage, it will finish
+     * when a segment reached it's size or time.
+     * Metadata passed in is for inspection purposes only and should be stored
+     * alongside the segment data.
+     *
+     * When the returned OffloaderHandle.getOffloadResultAsync completes, the corresponding
+     * ledgers has been persisted to the
+     * loadterm storage, so it is safe to delete the original copy in bookkeeper.
+     *
+     * The uid is used to identify an attempt to offload. The implementation should
+     * use this to deterministically generate a unique name for the offloaded object.
+     * This uid will be stored in the managed ledger metadata before attempting the
+     * call to offload(). If a subsequent or concurrent call to streamingOffload() finds
+     * a uid in the metadata, it will attempt to cleanup this attempt with a call
+     * to #deleteOffloaded(ReadHandle,UUID). Once the offload attempt completes,
+     * the managed ledger will update its metadata again, to record the completion,
+     * ensuring that subsequent calls will not attempt to offload the same ledger
+     * again.
+     *
+     * @return an OffloaderHandle, which when `completeFuture()` completed, denotes that the offload has been successful.
+     */
+    default CompletableFuture<OffloadHandle> streamingOffload(ManagedLedger ml, UUID uuid, long beginLedger,
+                                                              long beginEntry,
```

Comment:
> what happens if I call this method for the same ManagedLedger more times, with overlapping intervals?

Yes, the caller should prevent this happens.

Reply:
if the caller invokes this method with overlapping intervals, will the system be in a corrupted state ?
is there any way to detect the problem and fail ?

Reply:
ManagedLedger will check the begin entry and `endEntryId` in https://github.com/apache/pulsar/wiki/PIP-76:-Streaming-Offload#metadata-changes to avoid corruption, it's not included in this PR yet because it will make the PR too big and include too many details. This PR focused on the interface change of Offloader, I will submit another PR with corruption detection you mentioned.

Reply:
works for me, thanks